src/app/components/your-work/your-work.component.spec.ts

```diff
@@ -19,4 +68,28 @@ describe('YourWorkComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('render to do work table', () => {
+    fixture.detectChanges();
+    const compiled = fixture.nativeElement;
+    const title = compiled.querySelectorAll('.govuk-heading-m')[0];
+    expect(title).toBeTruthy();
+    expect(title.textContent).toEqual('To do');
+
+    const table = compiled.querySelector('#todoTable');
+    expect(table).toBeTruthy();
+    const tableRows = table.querySelectorAll('tr');
+    expect(tableRows.length).toEqual(2);
+    const firstRow = tableRows[1];
+    expect(firstRow).toBeTruthy();
+    const cells = firstRow.querySelectorAll('td');
+    expect(cells.length).toEqual(7);
+    expect(cells[0].textContent).toEqual('T2023453422');
+    expect(cells[1].textContent).toEqual('Reading');
+    expect(cells[2].textContent).toEqual('06 Aug 2023');
+    expect(cells[3].textContent).toEqual('Court Log');
+    expect(cells[4].textContent).toEqual('12 Aug 2023 13:00');
+    expect(cells[5].textContent).toEqual('Overnight');
+    expect(cells[6].textContent).toEqual('View');
+  });
```

Comment:
Can you add a test to cover the case with no data

Reply:
Test case added to cover no data being returned